hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/SlowPeerTracker.java

```diff
@@ -233,6 +234,20 @@ public String getSlowNode() {
     }
   }
 
+  /**
+   * Returns all tracking slow peers.
+   * @param numNodes
+   * @return
+   */
+  public ArrayList<String> getSlowNodes(int numNodes) {
+    Collection<ReportForJson> jsonReports = getJsonReports(numNodes);
+    ArrayList<String> slowNodes = new ArrayList<>();
+    for (ReportForJson jsonReport : jsonReports) {
+      slowNodes.add(jsonReport.getSlowNode());
+    }
+    return slowNodes;
+  }
```

Comment:
Somewhere in this method, we should log the slow peers as a WARN so that it shows up on Namenode logs.
This will be useful for admins when debugging HDFS performance issue. If they see list of slow nodes reported, they can quickly take a look at affected nodes instead of searching for various datanodes logs and searching for various kinds of "Slow Block Receiver" type of log messages in those logs.

Reply:
Good point.